src/plugins/usage_collection/server/routes/report_metrics.ts

```diff
@@ -48,15 +53,98 @@ export function registerUiMetricRoute(
     async (context, req, res) => {
       const { report } = req.body;
       try {
-        const internalRepository = getSavedObjects();
```

Comment:
i can get behind this as we can consume reports until it comes up. but devil's advocate:

do we want to keep this here incase saved objects client doesn't initialize?

Reply:
@kavilla We don’t exactly need the savedObjectClient for all the scenarios, we only require it for the scenarios where the batchingInterval has expired and we need to write the accumulated report into the savedObject. In rest of the cases we simply add it to the accumulated report. Moving the savedObjectClient initialization to the top might result in unnecessary failures. Given a scenario where the batchingInterval hasn’t expired but the savedObjectClient Initialization has failed, In the previous approach the report will be stored in the accumulator however in the updated scenario since there are client initialization failures the control moves to else block and  the report won’t be updated in the accumulator.